kernel/src/main/java/org/kframework/unparser/ColorSetting.java

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2014-2018 K Team. All Rights Reserved.
+package org.kframework.unparser;
+
+import java.util.Map;
+
+/**
+ * @author Denis Bogdanas
+ *         Date: 10/9/13
+ */
+public enum ColorSetting {
+    OFF, ON, EXTENDED;
+
+    private ColorSetting color;
+    static {
+        OFF.color      = OFF;
+        ON.color       = ON;
+        EXTENDED.color = EXTENDED;
+    }
+
+    public ColorSetting color(boolean ttyStdout, String outputFile, Map<String, String> env) {
+        boolean colorize = outputFile == null && ttyStdout;
+        ColorSetting colors = color;
```

Comment:
can't you just say this instead of color? and then delete the color field?

Reply:
I'm not sure if I know how to address this, but I will remove the intermediate variable `colors`, and just update the class-level `color` directly.

Reply:
I mean it looks to me like this code is setting an instance variable on each instance of the enumeration to itself. So `color == this` should always be true, which is why I don't think this field is necessary.

Reply:
Alright, I've re-done using `this`. Is this the correct way?